dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/netty/jfapchannel/NettyNetworkConnection.java

```diff
@@ -0,0 +1,330 @@
+					SibTr.debug(this, tc, "initChannel","Constructing pipeline");
+				bootstrap.handler(new ChannelInitializer<SocketChannel>() {
+					@Override
+					public void initChannel(SocketChannel ch) throws Exception {
+						final ChannelPipeline pipeline = ch.pipeline();
+						pipeline.addLast(NettyNetworkConnectionFactory.DECODER_HANDLER_KEY, new NettyToWsBufferDecoder());
+						pipeline.addLast(NettyNetworkConnectionFactory.ENCODER_HANDLER_KEY, new WsBufferToNettyEncoder());
+						pipeline.addLast(NettyNetworkConnectionFactory.HEARTBEAT_HANDLER_KEY, new JMSHeartbeatHandler(0));
+						pipeline.addLast(NettyNetworkConnectionFactory.JMS_CLIENT_HANDLER_KEY, new NettyJMSClientHandler());
+					}
+				});
+			}
+			
+			NettyNetworkConnection parent = this;
+			
+			
+			if(NettyNetworkConnectionFactory.USE_BUNDLE) {
+				nettyBundle.startOutbound(this.bootstrap, target.getRemoteAddress().getAddress().getHostAddress(), target.getRemoteAddress().getPort(), f -> {
+					if (f.isCancelled() || !f.isSuccess()) {
+						SibTr.debug(this, tc, "Channel exception during connect: " + f.cause().getMessage());
+						if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(parent, tc, "destroy", (Exception) f.cause());
+						listener.connectRequestFailedNotification((Exception) f.cause());
+						if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(parent, tc, "destroy");
+					}else {
+						if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(parent, tc, "ready", f);
+						parent.chan = f.channel();
+						listener.connectRequestSucceededNotification(readyConnection);
+						if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(parent, tc, "ready");
+					}
+
+				});
+			}else {
+				ChannelFuture oFuture = bootstrap.connect(target.getRemoteAddress());
+				oFuture.addListener(new ChannelFutureListener() {
+					
+					@Override
+					public void operationComplete(ChannelFuture f) throws Exception {
+						if (f.isCancelled() || !f.isSuccess()) {
+							SibTr.debug(this, tc, "Channel exception during connect: " + f.cause().getMessage());
+							if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(parent, tc, "destroy", (Exception) f.cause());
+							listener.connectRequestFailedNotification((Exception) f.cause());
+							if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(parent, tc, "destroy");
+						}else {
+							if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(parent, tc, "ready", f);
+							parent.chan = f.channel();
+							listener.connectRequestSucceededNotification(readyConnection);
+							if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(parent, tc, "ready");
+						}
+					}
+				});
+			}
+			
+			
+
+
+		} catch (Exception e) {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "destroy", e);
```

Comment:
I think you don;t wan't an entry and exit here (unless you add another return) and is "destroy" the method name?